tests/test-core/utils/js/ConditionalTestUtils.js

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+
+*/
+
+/* global fluid, jqUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.tests.conditionalTestUtils");
+
+    // "platformArg" is a string or array of strings of platform names to detect
+    // This returns a boolean based on whether or not the string is
+    // present in the navigator.platform string
+
+    fluid.tests.conditionalTestUtils.isPlatform = function (platformArg) {
+        var detectedPlatform = navigator.platform;
```

Comment:
This implementation should be rewritten so it can bind to any material configured by the Infusion ContextAwareness scheme rather than being hardwired to check for strings within navigator.platform (and this particular case of it bundled as ContextAwareness checks)